File: 02_구현_기출문제/02_문자열_재정렬.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main(void){

    string sort_s;
    string s;
    getline(cin, s);
    int sum = 0;

    for (auto i : s)
    {
        // 숫자라면
        if(i-'0' >= 0 && i-'0'<= 9){
            sum += i-'0';
        }
        else {
            sort_s.push_back(i);
        }
    }
    sort(sort_s.begin(), sort_s.end());

    cout << sort_s+ to_string(sum);

    return 0;
}

/*

K1KA5CB7

AJKDLSI412K4JSJ9D

*/

File: 02_구현_기출문제/02_문자열_재정렬.py
s = input()
flag = 0
sum = 0

result = []

for i in s :
    if i.isalpha() :
        result.append(i)
    else :
        flag = 1
        sum += int(i)

result.sort()

# 숫자가 있는 경우에만 뒤에 sum을 합친다
if flag :
    result.append(str(sum))

print(''.join(result))

'''

K1KA5CB7

AJKDLSI412K4JSJ9D

'''